tests/unittests/Foundation/NSStringTests.m

```diff
@@ -78,51 +78,54 @@ void testUrlCharacterSetEncoding(NSString* decodedString, NSString* encodedStrin
     allowedCharacterSet = [NSCharacterSet alphanumericCharacterSet];
 
     testString = [stringWillDie stringByAddingPercentEncodingWithAllowedCharacters:allowedCharacterSet];
-    ASSERT_OBJCEQ(urlEncodedString, testString);
+    EXPECT_OBJCEQ(urlEncodedString, testString);
 
     // stringByAppendingFormat test
     NSString* testString2 = @"Numbers: ";
     NSString* expectedString = @"Numbers: 1, 2";
     NSString* actualString = [testString2 stringByAppendingFormat:@"%d, %@", 1, @"2"];
-    ASSERT_OBJCEQ(expectedString, actualString);
+    EXPECT_OBJCEQ(expectedString, actualString);
 
     // stringWithFormat tests
     actualString = [NSString stringWithFormat:@"%x %d %u %.1f %o %c", 10, 11, 12, 13.0f, 14, 'a'];
-    ASSERT_OBJCEQ(@"a 11 12 13.0 16 a", actualString);
+    EXPECT_OBJCEQ(@"a 11 12 13.0 16 a", actualString);
 
     // %i is undocumented, but apps use it
     actualString = [NSString stringWithFormat:@"%i", -1];
-    ASSERT_OBJCEQ(@"-1", actualString);
+    EXPECT_OBJCEQ(@"-1", actualString);
 
     actualString = [NSString stringWithFormat:@"%hhd %hd %ld %llX", (char)1, (short)2, 3L, -1ULL];
-    ASSERT_OBJCEQ(@"1 2 3 FFFFFFFFFFFFFFFF", actualString);
+    EXPECT_OBJCEQ(@"1 2 3 FFFFFFFFFFFFFFFF", actualString);
 
+// size_t and ptrdiff_t are not a set size causing this to fail on the reference platform
+#ifdef WINOBJC
     actualString = [NSString stringWithFormat:@"%zx %zd", SIZE_MAX, INT_MIN];
-    ASSERT_OBJCEQ(@"ffffffff -2147483648", actualString);
+    EXPECT_OBJCEQ(@"ffffffff -2147483648", actualString);
 
     actualString = [NSString stringWithFormat:@"%tx %td", PTRDIFF_MAX, PTRDIFF_MIN];
-    ASSERT_OBJCEQ(@"7fffffff -2147483648", actualString);
+    EXPECT_OBJCEQ(@"7fffffff -2147483648", actualString);
+#endif
```

Comment:
hmm, is it worth adding an else here just so we can document how %z_ and %t_ behave on the reference platform?